packages/config/src/handlers/question/getQuestions/handler.ts

```diff
@@ -9,37 +9,100 @@
-import { internal } from '@hapi/boom'
+import { COUNTRY_CONFIG_URL } from '@config/config/constants'
+import FormDataset, {
+  IDataSetModel,
+  IDataset
+} from '@config/models/formDataset'
+import Question, { IQuestion } from '@config/models/question'
+
 import * as Hapi from '@hapi/hapi'
+import fetch from 'node-fetch'
+
+async function getQuestionsFromCountryConfig(
+  token: string
+): Promise<IQuestion[]> {
+  try {
+    const response = await fetch(`${COUNTRY_CONFIG_URL}/forms/questions`, {
+      headers: {
+        Authorization: token
+      }
+    })
+
+    if (response.status !== 200) {
+      return []
+    }
+
+    return response.json()
+  } catch (err) {
+    return []
+  }
+}
+
+type IQuestionFromCountryConfig = IQuestion & { datasetName?: string }
+
+function getOptionsForQuestion(
+  question: IQuestionFromCountryConfig,
+  datasets: IDataSetModel[]
+): IDataset['options'] | undefined {
+  if (!question.datasetId && !question.datasetName) {
+    return question.options
+  }
+
+  const dataset = datasets.find(
+    (dataset) =>
+      dataset._id === question.datasetId ||
+      dataset.fileName === question.datasetName
+  )
+
+  if (!dataset) {
+    return []
+  }
+
+  return dataset.options
+}
 
 export default async function getQuestions(
   request: Hapi.Request,
   h: Hapi.ResponseToolkit
 ) {
-  try {
-    const populatedQuestions = await Question.find()
-      .populate<{ datasetId: IDataSetModel }>('datasetId')
-      .exec()
-
-    const questions = populatedQuestions.map(async (populatedQuestion) => {
-      if (populatedQuestion.datasetId) {
-        const options = await resolveFormDatasetOptions(
-          populatedQuestion.datasetId
-        )
-        return {
-          ...populatedQuestion.toObject(),
-          options,
-          datasetId: populatedQuestion.datasetId._id
-        }
+  const datasets = await FormDataset.find().exec()
+
+  const allQuestionsFromCountryConfig = await getQuestionsFromCountryConfig(
+    request.headers.authorization
+  )
+
+  type QuestionWithDatasetPopulated = IQuestion & {
+    options?: IDataset['options']
+  }
+
+  /*
+   * Manually populate datasets for questions from country config
+   */
+
+  const questionsFromCountryConfig: QuestionWithDatasetPopulated[] =
+    allQuestionsFromCountryConfig.map((question) => {
+      return {
+        ...question,
+        options: getOptionsForQuestion(question, datasets)
       }
-      return populatedQuestion
     })
 
-    return Promise.all(questions)
-  } catch (error) {
-    throw internal(error.message)
-  }
+  /*
+   * Get all questions from form wizard and populate datasets
+   */
+  const questionsSavedInFormWizard = (await Question.find().exec()).map(
+    (question) => {
+      return {
+        ...question.toObject<IQuestion>(),
+        options: getOptionsForQuestion(question, datasets)
+      }
+    }
+  )
+
+  const allQuestions = questionsFromCountryConfig.concat(
```

Comment:
@rikukissa crazy question to think about, if you are merging these together, do we really need to deprecate form config UI?

Reply:
@euanmillar good question.. We haven't really tested the form config UI after these changes but in theory there's nothing stopping country config from defining the "base" form configuration and then form config UI extending it. So the hierarchy would be

1. Generic default.ts configurations in core
2. Country config /questions endpoint
3. Configuration made through form config UI

From a configuration process point of view, the UI configuration is of course quite tricky.. As we do not allow changing the form after there are records in the system, all config has to be done at the "development phase" of the system at which point it's important to be able to deploy the same form config to QA/Staging/Production environments. So changes someone makes through the UI in one of the environments would soon be overwritten. 

In my opinion the form config is also such complex and error-prone part of the system that I would not want anyone to be able to change it without code review and testing. 

In terms of codebase maintenance it would be brilliant if country config would just be able to provide default.ts type configuration and we could get rid of the configuration logic altogether. It's seriously one of the most complex parts of our frontend code and we have spent countless of hours trying to enable all the configuration options from country config we have today.

As a side note, we have been very happy with having an endpoint serving the form configuration. It makes iteration speed significantly faster. I would consider a similar approach for certificate templates.

Reply:
Yes, we are definitely going to deprecate it.  I was more asking because I wasn't sure if we would have time before the next release to remove and refactor all the related code and UI.